clang/lib/Headers/opencl-c-base.h

```diff
@@ -53,6 +53,13 @@
 #define __opencl_c_kernel_clock_scope_work_group 1
 #define __opencl_c_kernel_clock_scope_sub_group 1
 
+#ifdef __undef___opencl_c_integer_dot_product_input_4x8bit
```

Comment:
Perhaps move this down to the section that starts with the comment `// Undefine any feature macros that`...

Reply:
> Perhaps move this down to the section that starts with the comment `// Undefine any feature macros that`...

that section is guarded by OpenCL 3.0: `#if (__OPENCL_CPP_VERSION__ == 202100 || __OPENCL_C_VERSION__ == 300)`
But this section is guarded by OpenCL 2.0 and above.

Reply:
> Perhaps move this down to the section that starts with the comment `// Undefine any feature macros that`...

done in https://github.com/llvm/llvm-project/pull/162055/commits/4e77d69bb399bbe4d2bb87c4e4cf5e160a95d006